### DISCUSSION TITLE
What is needed to build libcurl with SFTP support?

### DISCUSSION THREAD

**Discussion by shuai555** [curl/curl#8051] [Q&A] [answered]
使用sftp协议传输文件，提示不支持。需要怎么解决

**Comment by jay** [comment, marked as answer]
Please communicate in English for help. (您需要用英语交流以获得讨论的帮助)

Your translated question is: _What is needed to build libcurl with SFTP support?_

SFTP support requires building libcurl with SSH library [libssh](https://www.libssh.org/), [libssh2](https://www.libssh2.org/) or [wolfSSH](https://www.wolfssl.com/products/wolfssh/). If you are using Linux your package manager may have a library already built. You can read more about how to build curl in the [everything curl book: build and install section](https://everything.curl.dev/source/build/fromsource).

**Reply by shuai555** [reply]
./configure  '--prefix=/home1/trunk_6/OPEN/libcurl/src/_install' '--host=arm-soft-linux-gnueabi' '--with-libssh2=PACKAGE/workshop/build/system/usr/lib/' 'CC=/home1/trunk_6/TOOLS/toolchain/arm-soft-linux-gnueabi/bin/arm-soft-linux-gnueabi-gcc' 'host_alias=arm-soft-linux-gnueabi'
configure: error: select TLS backend(s) or disable TLS with --without-ssl.

Select from these:

  --with-amissl
  --with-bearssl
  --with-gnutls
  --with-mbedtls
  --with-mesalink
  --with-nss
  --with-openssl (also works for BoringSSL and libressl)
  --with-rustls
  --with-schannel
  --with-secure-transport
  --with-wolfssl

**Reply by bagder** [reply]
@shuai555 you need to select which TLS backend you want to use. Or tell configure that you rather not use TLS at all (`--without-ssl`).

**Reply by shuai555** [reply]
The TLS backend I chose to use was libssh2, but an error occurred. Now I downloaded the libssh2 source code and compiled it into the libssh.so library. Stored in PACKAGE/workshop/build/system/usr/lib/, using'--with-libssh2=PACKAGE/workshop/build/system/usr /lib/ will prompt an error, what is the reason for this

**Reply by bagder** [reply]
libssh2 is an **SSH** library, used for SFTP and SCP. The **TLS** backend is separate.

**Comment by shuai555** [comment]
libssh2 is an SSH library, used for SFTP and SCP. The TLS backend is separate.What do I need to do to make curl support the sftp protocol

**Reply by bagder** [reply]
@jay [already answered that](https://github.com/curl/curl/discussions/8051#discussioncomment-1691566)

**Reply by shuai555** [reply]
I downloaded the source code of ssh2 and compiled it with the command ./configure && make -j128 && make install, why did not generate executable bin folder?

**Reply by ghost** [reply]
> I downloaded the source code of ssh2 and compiled it with the command ./configure && make -j128 && make install, why did not generate executable bin folder?

You are doing this very wrong. 
You should do these three steps separetely
Read `./configure --help` for config options
And why do you need 128 jobs for `make`?

**Comment by dfandrich** [comment]
I'm not aware of a project named ssh2. The libssh2 project, however, ships a
library, not an executable (a few demo apps notwithstanding). Unless you want
those demos, there won't be an executable to run. But, this isn't the right
forum for libssh2 questions.
